tests/unit-tests/src/VuFindTest/OaiPmh/CommunicatorTest.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+/**
+ * OAI-PMH harvester factory integration test.
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2010.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Demian Katz <demian.katz@villanova.edu>
```

Comment:
Feel free to give yourself credit for this if you wish!

Reply:
Will do. BTW, Institutional credit for all these fixes (including #8) also goes to Center for Research Libraries (www.crl.edu)